### PR TITLE
Improves support for Unix non-Linux systems, including QNX

### LINKED ISSUE #20395: Support for QNX Neutrino
##### System information (version)
- OpenCV => 4.5.3
- Operating System / Platform => Host: Windows 64 Bit / Target: QNX 7.1
- Compiler => QCC 7.1

##### Detailed description
Cross-compiling for QNX is not possible due to missing include headers in parallel due to lack of generalization for Unix systems.

##### Steps to reproduce
Build for any OS non-Linux system such as QNX, which is neither:
```
defined __linux__ || defined __APPLE__ || defined __GLIBC__ \
    || defined __HAIKU__ || defined __EMSCRIPTEN__ || defined __FreeBSD__ \
    || defined __OpenBSD__
```
See `parallel.cpp`.

##### Issue submission checklist

 - [x] I report the issue, it's not a question
   
 - [x] I checked the problem with documentation, FAQ, open issues,
       forum.opencv.org, Stack Overflow, etc and have not found solution
   
 - [x] I updated to latest OpenCV version and the issue is still there
   
 - [x] There is reproducer code and related data files: videos, images, onnx, etc

### PR DESCRIPTION
**Pull request by pablorcum**
Fixes #20395.

### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or other license that is incompatible with OpenCV
- [x] The PR is proposed to proper branch
- [x] There is reference to original bug report and related work
- [x] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [x] The feature is well documented and sample code can be built with the project CMake

```
force_builders=linux,docs,Android armeabi-v7a
```

### CODE REVIEW

File: modules/core/src/system.cpp [hunk 1]
```diff
@@ -114,10 +114,10 @@ void* allocSingletonNewBuffer(size_t size) { return malloc(size); }
 #include <cstdlib>        // std::abort
 #endif
 
-#if defined __ANDROID__ || defined __linux__ || defined __FreeBSD__ || defined __OpenBSD__ || defined __HAIKU__
+#if defined __ANDROID__ || defined __unix__ || defined __FreeBSD__ || defined __OpenBSD__ || defined __HAIKU__
 #  include <unistd.h>
 #  include <fcntl.h>
-#  include <elf.h>
+#  include <sys/elf.h>
```
Comment (alalek) [inline modules/core/src/system.cpp:120]: This change causes an build error on Linux x64 / Android ARM builds.

Comment (pablorcum) [reply modules/core/src/system.cpp:121]: I'll look into it.

Comment (pablorcum) [reply modules/core/src/system.cpp:121]: Fixed in cd306537f92676a741712511bfecd4336fa851cf.

**Review by alalek — APPROVED** [review]
Thank you :+1:

**Pull request merged**